src/gui/MixerView.cpp

```diff
@@ -185,11 +272,61 @@ int MixerView::addNewChannel()
 
 	updateMixerLine(newChannelIndex);
 
+	updateAutoTrackSortOrder();
 	updateMaxChannelSelector();
 
 	return newChannelIndex;
 }
 
+void MixerView::updateAutoTrackSortOrder()
+{
+
+	Mixer * mix = Engine::mixer();
+	QList<int> *list = new QList<int>();
+
+	// add all non auto track first
+	for( int i = 1; i<m_mixerChannelViews.size(); ++i )
+	{
+		if (!mix->mixerChannel(i)->m_autoTrackLinkModel.value())
+		{
+			list->append(i);
+		}
+	}
+
+	// add auto tracks in the order of the song tracks
+	mix->processAssignedTracks([&, list](Track * track, IntModel * model, MixerChannel * channel)
+	mutable {
+		(void) track;		
+		if (channel == NULL) return;		
+		if (channel->m_autoTrackLinkModel.value())
+		{
+			list->append(model->value());
+		}
+	});
+	return;
+
+
+	// bubblesort here because the list is normally almost ordered
```

Comment:
why not `std::sort` ?

Reply:
Because I am C++ beginner :-)

I will look into it.

Reply:
I was thinking of ``std::sort`` as well, but I was not sure how the default swap behavior would be changed to use ``swapChannels`` instead.

Reply:
Because it is described as a non stable sorting algorithm (https://en.cppreference.com/w/cpp/algorithm/sort) I could imagine that this is maybe a quicksort.

When I was thinking about the requirements I considered the following: the items are already nearly sorted, there are relatively few items and the sorting algorithm should swap in a mostly linear fashion. 

There is a good animation for comparison here:
https://www.toptal.com/developers/sorting-algorithms/nearly-sorted-initial-order

Of course it could be that there is in fact no difference if only one single element is out of order as it would be in this case.